shared/profile/user/weboftrust/container.tsx

```diff
@@ -1,44 +1,70 @@
 import * as Container from '../../../util/container'
-import * as Types from '../../../constants/types/profile'
+import * as UsersGen from '../../../actions/users-gen'
 import WebOfTrust from '.'
+import {WebOfTrustVerificationType} from '../../../constants/types/more'
+import {wotReactWaitingKey} from '../../../constants/users'
+import {WotReactionType, WotStatusType} from '../../../constants/types/rpc-gen'
 
 type OwnProps = {
   webOfTrustAttestation: {
     attestation: string
     attestingUser: string
-    dateString: string
-    pending: boolean
-    verificationType: Types.WebOfTrustVerificationType
+    status: WotStatusType
+    verificationType: WebOfTrustVerificationType
+    vouchedAt: number
   }
+  username: string
 }
 
 const Connected = Container.connect(
   (state, ownProps: OwnProps) => {
-    const {attestation, attestingUser, dateString, pending, verificationType} = ownProps.webOfTrustAttestation
+    const {username, webOfTrustAttestation} = ownProps
+    const {attestation, attestingUser, vouchedAt, status, verificationType} = webOfTrustAttestation
+    const userIsYou = ownProps.username === state.config.username
     return {
       attestation,
       attestingUser,
-      dateString,
-      pending,
-      username: state.config.username,
+      status,
+      userIsYou,
+      username,
       verificationType,
+      vouchedAt,
+    }
+  },
+
+  (dispatch: Container.TypedDispatch, {webOfTrustAttestation}: OwnProps) => {
+    const {status, attestingUser} = webOfTrustAttestation
+    switch (status) {
```

Comment:
Sounds like (at least so far) we could get away with just:
```js
        return {
          _onAccept: () => status === WotStatusType.proposed ?
            dispatch(UsersGen.createWotReact({reaction: WotReactionType.accept, voucher: attestingUser})) : null,
          _onReject: () =>
            dispatch(UsersGen.createWotReact({reaction: WotReactionType.reject, voucher: attestingUser})),
        }
```
rather than the switch, what do you think?

Reply:
yeah i was kind of building that out in expectation that i was going to do more of that work (different statuses exposing different functionality in this PR, but i opted to create follow-on tickets instead. i agree. will reduce.